### PR TITLE
Scripts for "New vehicle data WindowStatus" feature

### PR DESCRIPTION
**Pull request by GetmanetsIrina**
ATF Test Scripts to check #[3193 New vehicle data WindowStatus](https://github.com/smartdevicelink/sdl_core/issues/3193)

This PR is **ready** for review.

### Summary
Scripts for "New vehicle data WindowStatus" feature

### ATF version
develop

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/API/VehicleData/WindowStatus/023_OnVehicleData_notification_with_invalid_data.lua [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  lessThanMin = -2
+}
+
+local maxArraySize = {}
+for i = 1, invalidValue.greaterMax do
+  maxArraySize[i] = common.getWindowStatusParams()[1]
+end
+
+local invalidParam = {
+  ["empty_location"] = {
```
Comment (GetmanetsIrina) [inline test_scripts/API/VehicleData/WindowStatus/023_OnVehicleData_notification_with_invalid_data.lua:40]: @GetmanetsIrina , please add checks with missed approximatePosition or deviation.

Comment (GetmanetsIrina) [reply test_scripts/API/VehicleData/WindowStatus/023_OnVehicleData_notification_with_invalid_data.lua:40]: and also add checks for GetVD RPC

Comment (GetmanetsIrina) [reply test_scripts/API/VehicleData/WindowStatus/023_OnVehicleData_notification_with_invalid_data.lua:40]: @atiwari9, the update according to comments is performed in 3092bef5f0f3b3870272195f009bd15a5c0a8501

**Comment by GetmanetsIrina** [comment]
Please add one more script with checking of IGNORED resultCode in case app is already subscribed (also for UnsubscribeVD)

**Comment by GetmanetsIrina** [comment]
Scripts with resultCode IGNORED are added in 623f2eb

**Comment by atiwari9** [comment]
Test reports for commit: [623f2eb](https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2411/commits/623f2eb3c4f50032a37c7245f854aced7678abcb)

Logs: 
[WindowStatus_Final_Test_Report.zip](https://github.com/smartdevicelink/sdl_atf_test_scripts/files/4782854/WindowStatus_Final_Test_Report.zip)



```
=====================================================================================================
TOTAL: 34
PASSED: 28 
FAILED: 0 
ABORTED: 0 
SKIPPED: 6 
12 ;./test_scripts/API/VehicleData/WindowStatus/012_SubscribeVD_not_success_codes_for_windowStatus_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
19 ;./test_scripts/API/VehicleData/WindowStatus/019_UnsubscribeVD_not_success_codes_for_windowStatus_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
24 ;./test_scripts/API/VehicleData/WindowStatus/024_OnVehicleData_RPC_disallowed.lua https://github.com/smartdevicelink/sdl_core/issues/3365
25 ;./test_scripts/API/VehicleData/WindowStatus/025_OnVehicleData_disallowed_after_PTU.lua https://github.com/smartdevicelink/sdl_core/issues/3365
31 ;./test_scripts/API/VehicleData/WindowStatus/031_Resumption_of_subscription_for_2_apps_to_the_windowStatus_param_after_an_Unexpected_Disconnect.lua https://github.com/smartdevicelink/sdl_core/issues/2283
32 ;./test_scripts/API/VehicleData/WindowStatus/032_Resumption_of_subscription_for_2_apps_to_the_windowStatus_param_after_IGN_OFF_ON_cycle.lua https://github.com/smartdevicelink/sdl_core/issues/2283
=====================================================================================================
```

**Comment by LitvinenkoIra** [comment]
@atiwari9 please review this PR

**Comment by atiwari9** [comment]
Another round of tests: [623f2eb](https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2411/commits/623f2eb3c4f50032a37c7245f854aced7678abcb)

```
=====================================================================================================
TOTAL: 34
PASSED: 28 
FAILED: 0 
ABORTED: 0 
SKIPPED: 6 
12 ;./test_scripts/API/VehicleData/WindowStatus/012_SubscribeVD_not_success_codes_for_windowStatus_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
19 ;./test_scripts/API/VehicleData/WindowStatus/019_UnsubscribeVD_not_success_codes_for_windowStatus_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
24 ;./test_scripts/API/VehicleData/WindowStatus/024_OnVehicleData_RPC_disallowed.lua https://github.com/smartdevicelink/sdl_core/issues/3365
25 ;./test_scripts/API/VehicleData/WindowStatus/025_OnVehicleData_disallowed_after_PTU.lua https://github.com/smartdevicelink/sdl_core/issues/3365
31 ;./test_scripts/API/VehicleData/WindowStatus/031_Resumption_of_subscription_for_2_apps_to_the_windowStatus_param_after_an_Unexpected_Disconnect.lua https://github.com/smartdevicelink/sdl_core/issues/2283
32 ;./test_scripts/API/VehicleData/WindowStatus/032_Resumption_of_subscription_for_2_apps_to_the_windowStatus_param_after_IGN_OFF_ON_cycle.lua https://github.com/smartdevicelink/sdl_core/issues/2283
=====================================================================================================
```

Logs: 
[Test_Report_WindowStatus_623f2eb.zip](https://github.com/smartdevicelink/sdl_atf_test_scripts/files/4793189/Test_Report_WindowStatus_623f2eb.zip)

**Comment by atiwari9** [comment]
PR Approved.

**Comment by LitvinenkoIra** [comment]
@theresalech PR is ready for Livio review. Thank you!

**Comment by ShobhitAd** [comment]
@LitvinenkoIra In reference to https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2421#issuecomment-657482958. I think this vehicle data PR should also be updated based on https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2413#discussion_r451132043.

**Comment by ShobhitAd** [comment]
@LitvinenkoIra just as a note, in reference to https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2413#issuecomment-664457788, the updates to this PR should be based off of the `feature/vehicle_data_testing` branch.

**Comment by dboltovskyi** [comment]
@ShobhitAd Please notice as was agreed previously current PR has been completely reworked based on `feature/vehicle_data_testing` branch and is ready for PM review.

**Pull request merged**